### ISSUE TITLE
`QueryAssetRoots` RPC endpoints optionally filter for issuance only universe roots 

### ISSUE THREAD

**Issue by ffranr** [lightninglabs/taproot-assets#495]
TLDR: add functionality to multiverse sync only issuance proofs

---

The goal of this change is to facilitate issuance proof only multiverse federation syncing.

As things currently stand, multiverse synchronisation is prompted by comparing universe root hashes between local and remote nodes. The code doesn't currently allow us to exclusively synchronise issuance proofs. This is because each remote universe root hash that we compare may have been computed from a universe which is composed of a mix of issuance and transfer proof leaves.

A local node needs the ability to communicate to a remote node that it is interested in issuance proofs only. And, as a reply from the remote node, should receive back universe roots which were derived from issuance only universes.

In practice this means that the `QueryAssetRoots` RPC endpoint should be modified such that it is parameterised to optionally only return universe root(s) calculated from universe(s) composed of issuance only leaves.

**Comment by ffranr** [comment]
Depends on https://github.com/lightninglabs/taproot-assets/issues/494 which removes on of these two RPC endpoints.

**Comment by Roasbeef** [comment]
I think for multiverse syncing, we'd reformulate things entirely. At that point, we can't retry on the tree structure, so instead we want the ability to query _leaves_ that are only issuance or not otherwise. The alternative here is to maintain a set of in-memory shadow roots (creating the tree as if only issuance leaves were included), or even on disk. For now I think it's enough to tweak the syncing protocol slightly, rather than need to maintain roots for issuance vs otherwise. Though I think we'll want to do that eventually. 

Thoughts?

**Comment by ffranr** [comment]
@Roasbeef other options I think we should think about are:
* Internally storing two different multiverse trees: one for issuance only and one for transfer proofs. (Which I think might have been the original motivation for having **base** specific universe functionality.)
* Modify the MS-SMT tree such that we can pass in a custom leaf hashing function which would optionally ignore non-issuance leaves.

If we did the following three things we would be interface complete for multiverse issuance only syncing:
1. continue to sync based on tree root
2. add a flag to QueryAssetRoots (like this issue suggests)
3. add configuration to limit multiverse/universe to issuance only

Issuance only nodes in the federation would be limited to syncing with other issuance only nodes until we updated nodes to be able to generate issuance only tree roots from all-proofs multiverses.
